docs/classic-vs-vt.md

```diff
@@ -0,0 +1,115 @@
+
+While this provides additional power to Windows applications operating in a specific user-context on a single machine, it also provides a vector to cross security and privilege-levels or domains when used in certain scenarios. Such scenarios include operating between contexts on the same machine, or across contexts to another machine or environment.
+
+Other platforms, which use **virtual terminal sequences**, do not allow this activity. The intent of our recommendation to transition from classic Windows Console to virtual terminal sequences is to converge with this strategy for both interoperability and security reasons.
+
+### Direct Window Access
+
+**Windows Console API surface** provides the exact window handle to the hosting window. This allows a command-line utility to perform advanced window operations by reaching into the wide gamut of Win32 APIs permitted against a window handle. These Win32 APIs can manipulate the window state, frame, icon, or other properties about the window.
+
+By contrast, on other platforms with **virtual terminal sequences**, there is a narrow set of commands that can be performed against the window. These commands can do things like changing the window size or displayed title, but they must be done in the same band and under the same control as the remainder of the stream.
+
+As Windows has evolved, the security controls and restrictions on window handles have increased. Additionally, the nature and existence of an application-addressable window handle on any specific user interface element has evolved, especially with the increased support of device form factors and platforms. This makes direct window access to command-line applications fragile as the platform and experiences evolve.
+
+### Unicode
+
+The implicit standard for communication across platforms and the web is Unicode, specifically in the UTF-8 form. Of course, other encodings still exist.  However, when otherwise undefined, using UTF-8 is widely accepted as the appropriate default. It represents the balance of portability, storage/transmission size, and breadth of expression required to support the world's languages and glyphs.
+
+The **Windows Console** platform has supported and will continue to support all existing code pages and encodings, but we recommend UTF-8 for all forward looking development. We also accept UTF-16 as an algorithmically-translatable alternative.
+
+UTF-8 support in the console can be found via the _A_ variant of all Console APIs against console handles after setting the code page to `65001` or `CP_UTF8` with the [**SetConsoleOutputCP**](setconsoleoutputcp.md) and [**SetConsoleCP**](setconsolecp.md) methods, as appropriate. Setting the code pages in advance is only necessary if the machine has not chosen "Use Unicode UTF-8 for worldwide language support" in the settings for Non-Unicode applications in the Region section of the Control Panel.

```

Comment:
Which version of Windows is required for this? It doesn't seem to be the case for me on `Microsoft Windows [Version 10.0.19042.546]`. I have global UTF-8 enabled and also set both input/output console code pages to `65001` for good measure, yet both `ReadConsoleA()` and `ReadFile()` give me `NUL`s for non-ASCII characters such as `æøå` in all terminals I've tried.

Reply:
As far as I had been aware, `WriteConsoleA` and `ReadConsoleA` were working fine in `65001`. I'll make sure someone on the team double checks that before we publish this. Thanks for bringing it up.

If it isn't working, we'll change this language to talk about using UTF-16 for now as an algorithmic translation of UTF-8 and revisit once we work out whatever bug that is.

Reply:
OK. Here's the deal. I went and did a bunch of investigation and it looks like we still have work to do on making UTF-8 work in all scenarios for reading the input channel. I'll revise this text to say that our goal is to make UTF-8 work for everything and that it should already be working for the output stream, but the input stream will have varying levels of success until microsoft/terminal#7777 and friends are figured out. Hopefully that's alright for now.

Reply:
Sounds good. Thank you for investigating. 👍